core/domain/html_cleaner_test.py

```diff
@@ -784,3 +784,206 @@ def test_add_caption_to_image(self):
                 html_cleaner.add_caption_attr_to_image(
                     test_case['html_content']),
                 test_case['expected_output'])
+
+    def test_validate_customization_args(self):
```

Comment:
Do you have a test case with `oppia-noninteractive-tabs` (or collapsible) that contains another component inside of it (ie, nested components)?

Reply:
I had missed that before. I have added test now to ensure tabs and collapsible are not nested.

This also reminded me to add test for rich text components within tabs and collapsible like image in a tab component. I have done that too, PTAL